src/ansys/fluent/core/fluent_connection.py

```diff
@@ -89,6 +89,31 @@ def get_container(container_id_or_name: str) -> Union[bool, Container, None]:
     return container
 
 
+class ErrorState:
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def details(self):
+        return self._details
+
+    def __init__(self, name: str = "", details: str = ""):
+        self._name = name
+        self._details = details
+
+    def __eq__(self, other):
+        return self._name == other
```

Comment:
This version of `__eq__` should not be in the public interface. This version requires that RHS is required to be a string and not another object of the same type. `details` which is a public property is not involved in the comparison.